test/suites/proxy.sh

```diff
@@ -0,0 +1,78 @@
+test_proxy_device() {
+  
+  MESSAGE="Proxy device test string"
+
+  lxc launch busybox proxyTester
+  lxc config device add proxyTester proxyDev proxy listen=tcp:127.0.0.1:1234 connect=tcp:127.0.0.1:4321 bind=host
+  if [ $(lxc config device list proxyTester) != "proxyDev" ]; then
+    echo "Proxy device was not added to container"
+  fi
+
+
+  lxc config device remove proxyTester proxyDev
+  if [ $(lxc config device list proxyTester) ]; then 
+    echo "Proxy device was not removed from container"
+  fi
+
+  lxc config device add proxyTester proxyDev proxy listen=tcp:127.0.0.1:1234 connect=tcp:127.0.0.1:4321 bind=host
+  lxc stop proxyTester
+  if [ $(lxc config device list proxyTester) != "proxyDev" ]; then
+    echo "Proxy device should not be deleted from config on container stop"
+  fi
+  lxc delete proxyTester
+
+  lxc launch busybox proxyTester
+  if [ $(lxc config device list proxyTester) ]; then 
+    echo "Proxy device was not deleted from config on container deletion"
+  fi
+
+
+  lxc config device add proxyTester proxyDev proxy listen=tcp:127.0.0.1:1234 connect=tcp:127.0.0.1:4321 bind=host
+  nsenter -n -t $(lxc query /1.0/containers/proxyTester/state | jq .pid) -- nc -6 -l 4321 > proxyTest.out &
+  sleep 2
+
+  exec 3>/dev/tcp/localhost/1234
```

Comment:
You should be using the local_tcp_port instead of the hardcoded 1234 otherwise this test will be racy when run multiple times in parallel (as Jenkins does).

Reply:
While I was doing my initial testing, I was using `!#/bin/bash` when running my script. While doing some further testing, I realized that /dev/tcp/{host}/{port} does not work with the POSIX shell. The socket functionality we are trying to use is available in bash, but not the POSIX shell that the `main.sh` uses to run the tests. I am instead using `echo ${MESSAGE} | nc 127.0.0.1 1234 &` to send the data. When you said to use "local_tcp_port", did you mean to have a variable called `LOCAL_TCP_PORT` that I would declare at the top and use that instead of a hardcoded value or is there a port number that is conventionally used for these types of things?